### PR TITLE
refine complie configuration

### PR DESCRIPTION
**Pull request by jiaoew1991**
issue: https://github.com/milvus-io/milvus/issues/17495
Signed-off-by: Enwei Jiao <enwei.jiao@zilliz.com>

### CODE REVIEW

**Comment by jennyli-z** [comment]
This pr has increased docker image from 700MB to 1.47GB， double docker image size will cost double disk usage for CI nodes.
Currently we have downgrade the CI servers, and will move to the new cluster, hold for while before merged.

Better to figure out the reason why it increased ?

**Comment by jiaoew1991** [comment]
> This pr has increased docker image from 700MB to 1.47GB， double docker image size will cost double disk usage for CI nodes. Currently we have downgrade the CI servers, and will move to the new cluster, hold for while before merged.
> 
> Better to figure out the reason why it increased ?
It is because of new static library(about 800M) included in the image,  to run milvus does not require a static library. I have removed all static libraries from image to reduce size.

**Comment by jennyli-z** [comment]
> > This pr has increased docker image from 700MB to 1.47GB， double docker image size will cost double disk usage for CI nodes. Currently we have downgrade the CI servers, and will move to the new cluster, hold for while before merged.
> > Better to figure out the reason why it increased ?
> 
> It is because of new static library(about 800M) included in the image, to run milvus does not require a static library. I have removed all static libraries from image to reduce size. @jennyli-z
> 
> new image is about 600M image
Nice!

**Comment by XuanYang-cn** [comment]
@jiaoew1991 Great and complete job! Please resolve the conflicts so that I can label lgtm

/assign @xiaofan-luan

**Comment by XuanYang-cn** [comment]
/lgtm
@xiaofan-luan PTAL

**Comment by jiaoew1991** [comment]
after change rocksdb‘s directory, the golang-ci’s typecheck report lots of false alarms on windows.
could someone with windows OS help me. or can we skip static-check on windows ?

File: internal/core/src/storage/parquet_c.cpp [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+        st.error_code = static_cast<int>(ErrorCode::UNEXPECTED_ERROR);
+        st.error_msg = ErrorMsg("incorrect data type");
+        return st;
+    }
+    if (p->output != nullptr) {
+        st.error_code = static_cast<int>(ErrorCode::UNEXPECTED_ERROR);
+        st.error_msg = ErrorMsg("payload has finished");
+        return st;
+    }
+    auto ast = builder->AppendValues(reinterpret_cast<const uint8_t*>(values), length);
```
Comment (longjiquan) [inline internal/core/src/storage/parquet_c.cpp:291]: Maybe `static_cast` is more reasonable here than `reinterpret_cast`.

Comment (jiaoew1991) [reply internal/core/src/storage/parquet_c.cpp:291]: these files are moved from internal/storage/cwrapper, and formatted code.

File: Makefile [hunk 2]
```diff
@@ -19,19 +19,14 @@ LIBRARY_PATH := $(PWD)/lib
...
 	@echo "Running pre-processing"
-ifeq ($(OS),Darwin) # MacOS X
-	@echo "MacOS system identified. Switching to customized gorocksdb fork..."
-	@go mod edit -replace=github.com/tecbot/gorocksdb=github.com/soothing-rain/gorocksdb@v0.0.1
-endif
-ifeq ($(MSYSTEM), MINGW64) # MSYS
 	@echo "MSYS. Switching to customized gorocksdb fork..."
-	@go mod edit -replace=github.com/tecbot/gorocksdb=github.com/soothing-rain/gorocksdb@v0.0.1
-endif
+	@go mod edit -replace=github.com/tecbot/gorocksdb=github.com/jiaoew1991/gorocksdb@v0.0.2
```
[the remark this answers is not in the document]
Comment (jiaoew1991) [reply Makefile]: removed it

**Pull request merged**